Feat/20/push button driver

Eng-MohamedAbdelgaber opened the pull request:
@Hisham4Ahmed @AlaaE1

Eng-MohamedAbdelgaber committed:
[PushButton_Driver Version 1.0]
[
>>Provided a simple HAL driver for push button handling.
>>Detect button press using EXTI (falling edge trigger) .
>>Offer APIs for initialization, reading status, and ISR action handling.
>>Ensure debounce handling to avoid false triggers.

>>Button toggles reliably between Automatic ↔ Manual modes.
>>Interrupt triggers only on falling edge .
>>No spurious triggers due to bouncing.]
[Feat] #20

Commit:
[Review Points ]
[Check the .c file in the push button folder]
[Feat]   #20

Commit:
Doing last review point

Eng-MohamedAbdelgaber committed:
[PushButton_Driver Version 1.0]
[
**Enhanced Doxygen Comments
>>Provided a simple HAL driver for push button handling.
>>Detect button press using EXTI (falling edge trigger) .
>>Offer APIs for initialization, reading status, and ISR action handling.
>>Ensure debounce handling to avoid false triggers.

>>Button toggles reliably between Automatic ↔ Manual modes.
>>Interrupt triggers only on falling edge .
>>No spurious triggers due to bouncing.]
[Feat] #20

Eng-MohamedAbdelgaber committed:
Merge branch 'Feat/20/PushButton_Driver' of https://github.com/Hisham4Ahmed/Smart_Energy_Management_System into Feat/20/PushButton_Driver

Eng-MohamedAbdelgaber committed:
[PushButton_Driver Version 1.0]
[
** Enhanced Doxgen Comments
>>Provided a simple HAL driver for push button handling.
>>Detect button press using EXTI (falling edge trigger) .
>>Offer APIs for initialization, reading status, and ISR action handling.
>>Ensure debounce handling to avoid false triggers.

>>Button toggles reliably between Automatic ↔ Manual modes.
>>Interrupt triggers only on falling edge .
>>No spurious triggers due to bouncing.]
[Feat] #20

Eng-MohamedAbdelgaber committed:
[PushButton_Driver Version 1.0]
[
** Moved (extern uint8_t BTN_SystemMode) From Private to Interface.
** Enhanced Doxgen Comments
>>Provided a simple HAL driver for push button handling.
>>Detect button press using EXTI (falling edge trigger) .
>>Offer APIs for initialization, reading status, and ISR action handling.
>>Ensure debounce handling to avoid false triggers.

>>Button toggles reliably between Automatic ↔ Manual modes.
>>Interrupt triggers only on falling edge .
>>No spurious triggers due to bouncing.]
[Feat] #20

Hisham4Ahmed commented on Hal/PushButton/BTN_Interface.h:
```diff
@@ -0,0 +1,79 @@
+ * - 1 → Manual Mode  
+ * @note Declared here so that application (main) can read the current mode.
+*/
+extern uint8_t BTN_SystemMode;
```
This Global Var Shall be in Program.c and shall be static

Hisham4Ahmed commented on Hal/PushButton/BTN_Program.c:
```diff
@@ -0,0 +1,84 @@
+
+uint8_t hBtn_GetStatus(void)
+{
+    uint8_t BTNSTATE=0;
```
This function return the Global State

Eng-MohamedAbdelgaber committed:
[PushButton_Driver Version 2.0]
[

**Update Summary

**Resolved:** Issue reported by *Eng. Hesham* regarding the global variable linkage.
The variable `BTN_SystemMode` was previously declared as `extern`, which caused scope and access conflicts.
It has been modified to a **static** variable (`static uint8_t BTN_SystemMode = 0;`)
to ensure encapsulation within the `BTN_Program.c` file and prevent unwanted external access.
**Fixed:** The `hBtn_GetStatus()` function now correctly returns the `BTN_SystemMode` variable
instead of the button pin state as it is External interrupt Pushbutton not Normal Button driver so wee need to get the state of Currently BTN_SystemMode.
This update allows the application (`main.c`) to easily handle both **Automatic** and **Manual** modes.
-----------------------------------

Enhanced Doxgen Comments
>>Provided a simple HAL driver for push button handling.
>>Detect button press using EXTI (falling edge trigger) .
>>Offer APIs for initialization, reading status, and ISR action handling.
>>Ensure debounce handling to avoid false triggers.

>>Button toggles reliably between Automatic ↔ Manual modes.
>>Interrupt triggers only on falling edge .
>>No spurious triggers due to bouncing.]
[Feat] #20
